packages/app/utils/token.ts

```diff
@@ -1,8 +1,31 @@
+const getZeroDecimalsCount = (value: number | string) => {
+  if (!value.toString().includes(".")) return 0;
+
+  const firstDecimalZeroIndex = value
+    .toString()
+    .split(".")[1]
+    .split("")
+    .findIndex((digit: string) => digit >= "1");
+
+  return firstDecimalZeroIndex === -1 ? 0 : firstDecimalZeroIndex;
+};
+
 export const addressShortner = (address: string) =>
   address.slice(0, 4) + "..." + address.slice(-4);
 
-export const formatTokenValue = (value: number, custom?: number) => {
+export const formatTokenValue = (
+  value: number | string,
+  decimals: number = 4
+) => {
   if (value === 0) return value.toFixed(2);
 
-  return custom ? value.toFixed(custom) : value.toFixed(4);
+  const nonRelevantZeroes = getZeroDecimalsCount(value);
+  const relevantZeroDecimals = new Array(decimals - 1).fill(0).join("");
+
+  const formattedValue =
+    nonRelevantZeroes >= decimals
+      ? `< 0.${relevantZeroDecimals}1`
+      : Number(value).toFixed(decimals);
+
+  return formattedValue;
```

Comment:
```suggestion
  if (value === 0) return value;

  const fixedValue = (
    Math.floor(parseFloat(value as string) * 100) / 100
  ).toFixed(decimals);
  const parsedFixedValue = parseFloat(fixedValue);
  const relevantZeroDecimals = new Array(decimals - 1).fill(0).join("");

  const formattedValue =
    parsedFixedValue === 0
      ? `< 0.${relevantZeroDecimals}1`
      : Number(value).toFixed(decimals);

  return formattedValue;
```
I was suggesting something like this. We don't need to go through the array finding indexes. We can just cast it twice to know if it's lower than 0.0001.

Reply:
I've been doing a few runs with test values but at least on my end, it seems to not be working the same way. All these samples ran with 4 decimals (default for formatting for now) and different magnitudes:
* 0.001 (higher than 0.0001)
![image](https://github.com/SwaprHQ/stackly-ui/assets/21271189/d84e0688-d23d-4af8-8c01-6d8eed5eb35f)
* 0.0001 (equals to 0.0001)
![image](https://github.com/SwaprHQ/stackly-ui/assets/21271189/64a3aef0-c247-4ba2-9e34-a0011754b4de)
* 0.00001 (smaller than 0.0001, this result is ok)
![image](https://github.com/SwaprHQ/stackly-ui/assets/21271189/12f35fe6-2be6-4fde-a134-12336817c9ca)

Reply:
After an offline talk with Leo, we decided for a less tricky and simpler way to achieve this formatting.